packages/experience-builder-sdk/src/hooks/useFetchExperience.ts

```diff
@@ -144,21 +159,34 @@ export const useFetchExperience = ({ mode, client }: useClientsideExperienceFetc
           setExperience(experience);
 
           return experience;
-        } catch (e) {
-          handleError(errorMessagesWhileFetching.experienceReferences, e);
-          throw e;
+        } catch (error) {
+          handleError(errorMessagesWhileFetching.experienceReferences, error);
+          setError(error as Error);
         }
-      } catch (e) {
-        handleError(errorMessagesWhileFetching.experience, e);
-        throw e;
+      } catch (error) {
+        handleError(errorMessagesWhileFetching.experience, error);
+        setError(error as Error);
       } finally {
         setIsFetching(false);
       }
     },
     [client, mode]
   );
 
+  useEffect(() => {
+    if (slug) {
+      fetchBySlug({ slug, localeCode, experienceTypeId });
+      return;
+    }
+    if (id) {
+      fetchById({ id, localeCode, experienceTypeId });
+      return;
+    }
+    setError(Error('Either slug or id must be provided to useFetchExperience'));
+  }, [experienceTypeId, fetchById, fetchBySlug, id, localeCode, slug]);
+
   return {
+    error,
     fetchBySlug,
     fetchById,
```

Comment:
Aren't we now mixing 2 approaches together? If we commit to returning the `error` from the hook, then I would suggest instead returning `refetch` as a method, similar to `react-query` instead of the methods we currently return from the hook.

Separate functions are available via `import { fetchers } from '@contentful/experience-builder'` if needed.